File: abc/abc296/a.cpp
//問題:https://atcoder.jp/contests/abc296/tasks/abc296_a
//提出:https://atcoder.jp/contests/abc296/submissions/40209110
//2023-04-01AC
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin >> n;
    string s;
    cin >> s;
    bool ok = true;
    for(int i=1; i<n; ++i){
        if(s[i-1] == s[i]) ok = false;
    }
    if(ok) cout << "Yes" << endl;
    else cout << "No" << endl;
}

File: abc/abc296/a.rb
#問題:https://atcoder.jp/contests/abc296/tasks/abc296_a
#提出:https://atcoder.jp/contests/abc296/submissions/40295651
#2023-04-02AC
n = gets.to_i
s = gets
for i in 0.. n-2
    if s[i] == s[i+1]
        puts "No"
        exit
    end 
end
puts "Yes"